pkg/types/vsphere/platform.go

```diff
@@ -72,4 +107,207 @@ type Platform struct {
 	// specified, it will be set according to the default storage policy
 	// of vsphere.
 	DiskType DiskType `json:"diskType,omitempty"`
+
+	// vcenters holds the connection details for services to communicate with vCenter.
+	// Currently only a single vCenter is supported.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:validation:MinItems=1
+	VCenters []VCenterSpec `json:"vcenters,omitempty"`
+
+	// vSphere location where openshift rhcos virtual machines will be deployed
+	// based on vCenter and failureDomain
+	// If this is omitted failure domains (regions and zones) will not be used.
+	// +kubebuilder:validation:Optional
+	DeploymentZones []DeploymentZoneSpec `json:"deploymentZones,omitempty"`
+
+	// failureDomains holds the VSpherePlatformFailureDomainSpec which contains
+	// the definition of region, zone and the vCenter topology.
+	// If this is omitted failure domains (regions and zones) will not be used.
+	// +kubebuilder:validation:Optional
+	FailureDomains []FailureDomainSpec `json:"failureDomains,omitempty"`
+}
+
+// FailureDomain holds the name of the associated tag, the type
+// of the failure domain, and the vCenter tag category associated with this
+// failure domain.
+type FailureDomain struct {
+	// name is the name of the vCenter tag that represents this failure domain
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	Name string `json:"name"`
+
+	// type is the name of the failure domain type, which includes
+	// Datacenter, ComputeCluster and HostGroup
+	// +kubebuilder:validation:Enum=HostGroup;Datacenter;ComputeCluster
+	// +kubebuilder:validation:Required
+	Type FailureDomainType `json:"type"`
+
+	// tagCategory is the category used for the tag
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	TagCategory string `json:"tagCategory"`
+}
+
+// FailureDomainSpec holds the region and zone failure domain and
+// the vCenter topology of that failure domain.
+type FailureDomainSpec struct {
```

Comment:
I left a comment on the enhancement about this, because it seems more appropriate to discuss there: https://github.com/openshift/enhancements/pull/918/files#r933507865

But generally I would lean in favor of removing the `Spec` suffix from all of these variable names.